src/Logic.hs

```diff
@@ -266,10 +266,12 @@ handlePullRequestCommitChanged pr newSha state =
       Nothing -> pure state
 
 handlePullRequestClosed :: PullRequestId -> ProjectState -> Action ProjectState
-handlePullRequestClosed pr state = return $ Pr.deletePullRequest pr state {
-  -- If the PR was the current integration candidate, reset that to Nothing.
-  Pr.integrationCandidate = mfilter (/= pr) $ Pr.integrationCandidate state
-}
+handlePullRequestClosed pr state = Pr.deletePullRequest pr <$>
+    case Pr.integrationCandidate state of
+      Just candidatePr | candidatePr == pr -> do
+        leaveComment pr "Abandoning this pull request because it was closed."
+        return state { Pr.integrationCandidate = Nothing }
```

Comment:
We prefer to write `pure` instead of return, both because it is from applicative after all, and because it is less confusing for newcomers who have an expectation of what `return` does from an imperative language.